src/openrct2/peep/Guest.cpp

```diff
@@ -1583,30 +1585,32 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
         if (itemValue < price)
         {
             itemValue -= price;
-            if (shopItem == SHOP_ITEM_UMBRELLA)
+            if (isRainingAndUmbrella)
             {
-                if (climate_is_raining())
-                    goto loc_69B221;
+                // do nothing for now
             }
 
-            itemValue = -itemValue;
-            if (Happiness >= 128)
-                itemValue /= 2;
+            if (!isRainingAndUmbrella)
+            {
+                itemValue = -itemValue;
+                if (Happiness >= 128)
+                    itemValue /= 2;
 
-            if (Happiness >= 180)
-                itemValue /= 2;
+                if (Happiness >= 180)
+                    itemValue /= 2;
 
-            if (itemValue > (static_cast<money16>(scenario_rand() & 0x07)))
-            {
-                // "I'm not paying that much for x"
-                PeepThoughtType thought_type = static_cast<PeepThoughtType>(
-                    (shopItem >= 32 ? (PEEP_THOUGHT_TYPE_PHOTO2_MUCH + (shopItem - 32))
-                                    : (PEEP_THOUGHT_TYPE_BALLOON_MUCH + shopItem)));
-                InsertNewThought(thought_type, ride->id);
-                return false;
+                if (itemValue > (static_cast<money16>(scenario_rand() & 0x07)))
+                {
+                    // "I'm not paying that much for x"
+                    PeepThoughtType thought_type = static_cast<PeepThoughtType>(
+                        (shopItem >= 32 ? (PEEP_THOUGHT_TYPE_PHOTO2_MUCH + (shopItem - 32))
+                                        : (PEEP_THOUGHT_TYPE_BALLOON_MUCH + shopItem)));
+                    InsertNewThought(thought_type, ride->id);
+                    return false;
+                }
             }
         }
-        else
+        else if (!isRainingAndUmbrella)
```

Comment:
drop this extra if and I think you should pass the tests.

Reply:
I missed it! Thanks so much!